### PR TITLE
Show correct Hardware wallet errors for transactions 

### LINKED ISSUE #4882: Implement hardware wallet device status detection
### Description

The current implementation of knowing if the device is connected/disconnected/standby causes the device to slowdown due to too many requests and not optimised the implementation.

### Acceptance Criteria

- Implement hardware wallet status functionality
- Handle event connection, disconnection and standby
  - Connection and disconnection is identifiable through events, however standby status is hard to detect
- Implement all the category of error status
  - Handle possible error status i.e (open, close)
- Add unit tests for status updates

### PR DESCRIPTION
**Pull request by oskarleonard**
### What was the problem?

This PR resolves #4882
Review [this PR](https://github.com/LiskHQ/lisk-desktop/pull/4985) before this one.

### How was it solved?

- Adding 3 error state (disconnected, rejected, Lisk app closed) for hardware wallet when initialing a transaction.
- Added UX to prevent user from initialing a transaction when we know it wont work.
- Removed dead code

### How was it tested?
Manually these 3 error states:
1. Connect HW device, enter pin code, dont open the Lisk app. Try to make a transaction. 
   - Expected: Send button to be disabled and show error label with "Open the Lisk app to continue". Open Lisk app on device, send button should now be enabled. 
2. Enter send flow when "Confirm your transaction" modal pops up, disconnect the device.
   - Expected: A modal showing "Device disconnected"
3. Enter send flow when "Confirm your transaction" modal pops up, reject the transaction on the device.
   - Expected: A modal showing "Transaction aborted on device"

### CODE REVIEW

**Review by ManuGowda — CHANGES_REQUESTED** [review]
Please review and confirm the transaction on your ledger device.

**Comment by ManuGowda** [comment]
please address this comment as well https://github.com/LiskHQ/lisk-desktop/pull/4985#discussion_r1193759409

File: src/modules/transaction/components/TxSummarizer/footer.js [hunk 1]
```diff
@@ -44,9 +48,17 @@ const Footer = ({ confirmButton, cancelButton, footerClassName, t }) => {
 
   return (
     <BoxFooter className={`${footerClassName} summary-footer`} direction="horizontal">
+      {isHW && !isAppOpen && (
+        <div className={styles.errorLabel}>
+          <span>{t('Open the Lisk app to continue')}</span>
```
Comment (ManuGowda) [inline src/modules/transaction/components/TxSummarizer/footer.js]:
```suggestion
          <span>{t('Open the Lisk app on Ledger device to continue')}</span>
```

File: src/modules/transaction/components/TxBroadcaster/illustrationsMap.js [hunk 2]
```diff
@@ -1,27 +1,32 @@
...
   },
   stake: {
     [txStatusTypes.signatureSuccess]: 'stakingSuccess',
     [txStatusTypes.broadcastSuccess]: 'stakingSuccess',
     [txStatusTypes.broadcastError]: 'transactionError',
     [txStatusTypes.signatureError]: 'transactionError',
-    [txStatusTypes.hwRejected]: 'HwRejection',
+    [txStatusTypes.hwRejected]: 'hwRejection',
+    [txStatusTypes.hwDisconnected]: 'hwRejection',
+    [txStatusTypes.hwLiskAppClosed]: 'hwRejection',
```
Comment (ManuGowda) [inline src/modules/transaction/components/TxBroadcaster/illustrationsMap.js:20]: for all types we are classifying them as `hwRejection`?

**Review by ManuGowda — CHANGES_REQUESTED** [review]

File: src/modules/transaction/utils/hwManager/index.js [hunk 3]
```diff
@@ -17,19 +17,14 @@ const signTransactionByHW = async ({ wallet, schema, chainID, transaction }) =>
...
-    throw new Error(error);
-  }
+  const signature = await getSignedTransaction(
+    wallet.hw.path,
+    wallet.metadata.accountIndex,
+    unsignedMessage
+  );
+  transaction.signatures.push(signature.signature);
+  return transaction;
+  // return updateTransactionSignatures(wallet, transaction, signature);
```
Comment (eniolam1000752) [inline src/modules/transaction/utils/hwManager/index.js:27]: 💣

**Review by eniolam1000752 — CHANGES_REQUESTED** [review]

File: src/modules/transaction/components/TxBroadcaster/illustrationsMap.js (hunk 2 above)
Comment (oskarleonard) [reply src/modules/transaction/components/TxBroadcaster/illustrationsMap.js:20]: Each hw tx error has its own `txStatusType`, however, all use the same error icon

File: src/modules/transaction/utils/hwManager/index.js (hunk 3 above)
Comment (oskarleonard) [reply src/modules/transaction/utils/hwManager/index.js:27]: @ManuGowda can we safely delete this function? I think you left it commented out when you refactored the multi signature part.

Comment (ManuGowda) [reply src/modules/transaction/utils/hwManager/index.js:27]: ooo.. the multi signature is working for wallet now, it should be same for hardware wallet, did you test registering a multisign on hw and signing it?

File: src/modules/transaction/components/TxBroadcaster/illustrationsMap.js (hunk 2 above)
Comment (ManuGowda) [reply src/modules/transaction/components/TxBroadcaster/illustrationsMap.js:20]: hmmm ok.. @Ade-kola could have given different icons if needed

**Review by ManuGowda — APPROVED** [review]

**Review by eniolam1000752 — APPROVED** [review]

**Pull request merged**